src/simsopt/geo/magneticfieldclasses.py

```diff
@@ -0,0 +1,116 @@
+        self.dBlambdify_by_dX = sp.lambdify((R, Z, Phi), 
+                            [[sp.cos(Phi)*self.Phiparsed.diff(R).diff(R)-(sp.sin(Phi)/R)*self.Phiparsed.diff(R).diff(Phi),sp.cos(Phi)*(self.Phiparsed.diff(Phi)/R).diff(R)-(sp.sin(Phi)/R)*(self.Phiparsed.diff(Phi)/R).diff(Phi),sp.cos(Phi)*self.Phiparsed.diff(Z).diff(R)-(sp.sin(Phi)/R)*self.Phiparsed.diff(Z).diff(Phi)],
+                             [sp.sin(Phi)*self.Phiparsed.diff(R).diff(R)+(sp.cos(Phi)/R)*self.Phiparsed.diff(R).diff(Phi),sp.sin(Phi)*(self.Phiparsed.diff(Phi)/R).diff(R)+(sp.cos(Phi)/R)*(self.Phiparsed.diff(Phi)/R).diff(Phi),sp.sin(Phi)*self.Phiparsed.diff(Z).diff(R)+(sp.cos(Phi)/R)*self.Phiparsed.diff(Z).diff(Phi)],
+                             [self.Phiparsed.diff(R).diff(Z),(self.Phiparsed.diff(Phi)/R).diff(Z),self.Phiparsed.diff(Z).diff(Z)]])
+        
+    def compute(self, points, compute_derivatives=0):
+        r   = np.sqrt(np.power(points[:,0],2) + np.power(points[:,1],2))
+        z   = points[:,2]
+        phi = np.arctan2(points[:,1],points[:,0])
+        self._B = [self.Blambdify(r[i],z[i],phi[i]) for i in range(len(r))]
+
+        if compute_derivatives >= 1:
+            self._dB_by_dX = [self.dBlambdify_by_dX(r[i],z[i],phi[i]) for i in range(len(r))]
+
+
+class CircularCoilXY(MagneticField):
+    '''Magnetic field created by a single circular coil in the xy plane evaluated using analytical functions, including complete elliptic integrals of the first and second kind.
+    As inputs, it takes the radius of the coil (r0), its center and current (I).
+    Note: this function needs scipy.
+
+    Args:
+        r0: radius of the coil
+        center: point at the coil center
+        I: current of the coil in Ampere's
+    '''
+    def __init__(self, r0=0.1, center=[0,0,0], I=5e5/np.pi):
+        self.r0     = r0
+        self.Inorm  = I*4e-7
+        self.center = center
+
+    def compute(self, points, compute_derivatives=0):
+        points = np.array([np.subtract(point,self.center) for point in points])
+        rho   = np.sqrt(np.power(points[:,0],2) + np.power(points[:,1],2))
+        r     = np.sqrt(np.power(points[:,0],2) + np.power(points[:,1],2) + np.power(points[:,2],2))
+        alpha = np.sqrt(self.r0**2 + np.power(r,2) - 2*self.r0*rho)
+        beta  = np.sqrt(self.r0**2 + np.power(r,2) + 2*self.r0*rho)
+        k     = np.sqrt(1-np.divide(np.power(alpha,2),np.power(beta,2)))
+        self._B = np.array([
+            [self.Inorm*point[0]*point[2]/(2*alpha[i]**2*beta[i]*rho[i]**2)*((self.r0**2+r[i]**2)*ellipe(k[i]**2)-alpha[i]**2*ellipk(k[i]**2)),
+             self.Inorm*point[1]*point[2]/(2*alpha[i]**2*beta[i]*rho[i]**2)*((self.r0**2+r[i]**2)*ellipe(k[i]**2)-alpha[i]**2*ellipk(k[i]**2)),
+             self.Inorm/(2*alpha[i]**2*beta[i])*((self.r0**2-r[i]**2)*ellipe(k[i]**2)+alpha[i]**2*ellipk(k[i]**2))]
+            for i,point in enumerate(points)])
+
+## Next magnetic field classes to implement
```

Comment:
This should not be in the source code. Maybe open an issue if you want to keep track of our todos

Reply:
Issue opened